syntax = "proto3";

package Group;

option cc_generic_services = true;

message GroupUser
{
    bytes name = 1;
    bytes email = 2;
    bytes phone = 3;
    bytes role = 4;
}

message GroupInfo
{
    int32 id = 1;
    bytes name = 2;
    bytes desc = 3;
    repeated GroupUser users = 4;
}

message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message CreateGroupRequest
{
    GroupInfo Group = 1;
}

message CreateGroupResponse
{
    ResultCode result = 1;
    bool success = 2;
}

message AddGroupRequest
{
    bytes adminName = 1;
    bytes groupName = 2;
    bytes role = 3;
}

message AddGroupResponse
{
    ResultCode result = 1;
    bool success = 2;
}

message QueryGroupRequest
{
    bytes adminName = 1;
}

message QueryGroupResponse
{
    ResultCode result = 1;
    bool success = 2;
    repeated GroupInfo groupInfo = 3;
}

message QueryGroupUsersRequest
{
    bytes adminName = 1;
    bytes groupName = 2;
}

message QueryGroupUsersResponse
{
    ResultCode result = 1;
    bool success = 2;
    repeated bytes groupUserName = 3;
}

service GroupServiceRpc
{
    rpc CreateGroup(CreateGroupRequest)returns(CreateGroupResponse);
    rpc AddGroup(AddGroupRequest)returns(AddGroupResponse);
    rpc QueryGroup(QueryGroupRequest)returns(QueryGroupResponse);
    rpc QueryGroupUsers(QueryGroupUsersRequest)returns(QueryGroupUsersResponse);
}